spec.md

```diff
@@ -703,6 +704,36 @@ Probe types may target `const`{.firrtl} signals, but cannot use
 `RWProbe<const T>`{.firrtl}, as constant values should never be mutated at
 runtime.
 
+## Type Alias
+
+A type alias provides a way to assign names to existing FIRRTL types, enabling
+their reuse across multiple declarations and potentially serving as hints in
+verilog code generation.
+
+```firrtl
+type WordType = UInt<32>
+type ValidType = UInt<1>
+type Data = {w: WordType, valid: ValidType, flip ready: UInt<1>}
+type AnotherWordType = UInt<32>
+
+module TypeAliasMod:
+  input in: Data
+  output out: Data
+  wire w: AnotherWordType
+  w <= in.w
+  ...
+```
+
+The `type` declaration is globally defined and all named types exist in the same
+namespace and thus must all have a unique name. Type alias doesn't share the same
+type scope as modules; hence it is allowed for type aliaes to conflict with module
+names. In the output verilog the given names may not be preserved by the FIRRTL
+compiler, as names must be unique in the output relative to target-dependent scoping
+rules which may require changing names or dropping. Note that type aliases do not
+introduce nominal types unlike bundle field names; the type equivalence is determined
+solely by their structure. For instance types of `w`{.firrtl} and `in.w`{.wirrtl} are
+equivalent even though they are different type alias.
```

Comment:
The PR could avoid talking about "nominal types" and just say that the types are compared structurally.  This can then refer to the type equivalence section for more information.